ext/sodium/libsodium.c

```diff
@@ -2071,24 +2003,18 @@ PHP_FUNCTION(sodium_crypto_aead_aes256gcm_decrypt)
 		RETURN_THROWS();
 	}
 	if (npub_len != crypto_aead_aes256gcm_NPUBBYTES) {
-		zend_throw_exception(sodium_exception_ce,
-				   "public nonce size should be "
-				   "SODIUM_CRYPTO_AEAD_AES256GCM_NPUBBYTES bytes",
-				   0);
+		zend_argument_error(sodium_exception_ce, 3, "to be SODIUM_CRYPTO_AEAD_AES256GCM_NPUBBYTES bytes long");
 		RETURN_THROWS();
 	}
 	if (secretkey_len != crypto_aead_aes256gcm_KEYBYTES) {
-		zend_throw_exception(sodium_exception_ce,
-				   "secret key size should be "
-				   "SODIUM_CRYPTO_AEAD_AES256GCM_KEYBYTES bytes",
-				   0);
+		zend_argument_error(sodium_exception_ce, 4, "to be SODIUM_CRYPTO_AEAD_AES256GCM_KEYBYTES bytes long");
 		RETURN_THROWS();
 	}
 	if (ciphertext_len < crypto_aead_aes256gcm_ABYTES) {
 		RETURN_FALSE;
 	}
 	if (ciphertext_len - crypto_aead_aes256gcm_ABYTES > 16ULL * ((1ULL << 32) - 2ULL)) {
-		zend_throw_exception(sodium_exception_ce, "message too long for a single key", 0);
+		zend_argument_error(sodium_exception_ce, 1, "to have a smaller size");
```

Comment:
The `"to be shorter than the maximum allowed length"` formulation that you use elsewhere sounds better imho...

Reply:
Fixed!